File: lookandsay.rb
#!/usr/bin/env ruby -wKU
# lookandsay.rb
#   juanfc 2010-04-15
# 

class String
  def look_and_say
    gsub(/(.)\1*/){|s| "#{s.size}#{s[0,1]}"}
  end
end

s = '1'
resultados= [s]

timer= Time.now
(24).times {
  resultados.push(s = s.look_and_say)
}
timer= (Time.now- timer) * 1e3

p resultados.shift until resultados.length == 0
puts(timer.to_s + " milisegundos.")

File: lookAndSay.js
#!/usr/bin/env node
//JavaScript is the Best
//20100415 jorge@jorgechamorro.com

var puts= require("sys").puts;

Number.prototype.times= function (f) {
  var n= +this;
  while (n--) f(n);
};


String.prototype.lookAndSay= function () {
  var r= "";
  this.match(/(.)\1*/g).forEach(function (s) { r+= s.length+ s[0]; });
  return r;
};

var s= "1";
var resultados= [s];

var timer= +new Date();
(24).times(function () {
  resultados.push(s= s.lookAndSay());
});
timer= new Date()-timer;

while (resultados.length) puts(resultados.shift());
puts(timer+ " milisegundos.");